### ISSUE TITLE
Using named schema in items field breaks schema validation

### ISSUE THREAD

**Issue by SkeLLLa** [fastify/fastify#1491]
## 🐛 Bug Report

Using some named schema as `items` in other schema cause exception when using first schema

## To Reproduce

Add schema with simple item. Then add schema for items array and use first schema `$id` in `items`.
Then use the first schema in some route.

Paste your code here:

```js
const fastify = require('fastify')();

(async () => {
  const name = 'item';
  const item = {type: 'object', properties: {foo: {type: 'string'}}};
  fastify.addSchema({
    $id: `${name}`,
    ...item,
  });
  fastify.addSchema({
    $id: `${name}List`,
    type: 'array',
    items: `${name}#`,
  });
  fastify.get(
    '/',
    {
      schema: {
        response: {
          200: `${name}#`,
        },
      },
    },
    async (req, res) => {
      return {};
    }
  );
  await fastify.ready();
  await fastify.listen(3000);
  console.log('app listens');
})();

```

```
(node:21665) UnhandledPromiseRejectionWarning: Error: schema is invalid: data.items should be object,boolean, data.items should be array, data.items should match some schema in anyOf
at Ajv.validateSchema (/tmp/xxx/node_modules/ajv/lib/ajv.js:177:16)
at Ajv._addSchema (/tmp/xxx/node_modules/ajv/lib/ajv.js:306:10)
at Ajv.addSchema (/tmp/xxx/node_modules/ajv/lib/ajv.js:136:29)
at Object.keys.forEach.key (/tmp/xxx/node_modules/fast-json-stringify/index.js:937:11)
at Array.forEach (<anonymous>)
at isValidSchema (/tmp/xxx/node_modules/fast-json-stringify/index.js:936:33)
at build (/tmp/xxx/node_modules/fast-json-stringify/index.js:39:3)
at getValidatorForStatusCodeSchema (/tmp/xxx/node_modules/fastify/lib/validation.js:13:10)
at /tmp/xxx/node_modules/fastify/lib/validation.js:19:21
at Array.reduce (<anonymous>)
```

## Expected behavior

No exception thrown.

## Your Environment

- *node version*: 11
- *fastify version*: 2.0.0
- *os*: Mac, Linux

**Comment by mcollina** [comment]
@SkeLLLa thanks for reporting! Would you like to send a PR to fix it?

Also @Eomm might now more.

**Comment by SkeLLLa** [comment]
@mcollina I don't know exactly where exactly this should be. Probably even in `fast-json-stringify`, so may be it will be better if someone who is more familiar with both `fastify` and `fast-json-stringify` libs.

**Comment by Eomm** [comment]
I will check this evening 👍
The problem seems in the cross-shared-schema references

**Comment by Eomm** [comment]
This bug occurs only if you don't use a shared schema.

If you add the shared schema in a body, for example, all works.

```js
// workaround
  fastify.get('/', {
      schema: {
        body: `${name}List#`,
        response: {
          200: `${name}#`
        }
      }
    },
```

The "replace-way" shared schemas are objects that are modified: the `key: <id>#` part is replaced with a JSON resulting `key: {<JSON>}`.
This update of the reference is triggered only when Fastify start the loading.

But, if you have a shared schema that references a not used shared schema, it will be not evaluated but it will be pass to serializer here:

https://github.com/fastify/fastify/blob/c24b14a5e5e0b3bcc1bbe63f3a4a7c5476190e56/lib/validation.js#L53

@SkeLLLa , for what I tested it seems that the bug is also in v1, could you share your test?

PR with the fix is coming 👍

**Comment by SkeLLLa** [comment]
@Eomm I've tested it with code from the first post. And it worked in v1.
And unfortunately workaround will not work well in my case, because I need `POST` request as well :).

**Comment by Eomm** [comment]
Fix released in [v2.0.1](https://github.com/fastify/fastify/releases/tag/v2.0.1)
Thank you for reporting ✌
